src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,296 @@
+     * @param what The changed parameter name. `"quitTo", "to"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            require(data != address(0), "RwaInputConduit3/invalid-quit-to-address");
+            quitTo = data;
+        } else if (what == "to") {
+            require(data != address(0), "RwaInputConduit3/invalid-to-address");
+            to = data;
+        } else {
+            revert("RwaInputConduit3/unrecognised-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Swaps the GEM balance of this contract into DAI through the PSM and push it into the `to` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function push() external isMate {
+        _doPush(gem.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Swaps the specified amount of GEM into DAI through the PSM and push it into the `to` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param amt Gem amount.
+     */
+    function push(uint256 amt) external isMate {
+        _doPush(amt);
+    }
+
+    /**
+     * @notice Flushes out any GEM balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external isMate {
+        _doQuit(gem.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out specific amount of GEM balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param amt Gem amount.
+     */
+    function quit(uint256 amt) external isMate {
+        _doQuit(amt);
+    }
+
+    /**
+     * @notice Flushes out all outstanding DAI balance to `usr` address.
+     * @dev Can only be called by the admin
+     * @param usr Destination address.
+     */
+    function quitDai(address usr) external auth {
+        dai.transfer(usr, dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Calculate amount of DAI received for swapping GEM through PSM.
+     * @param gemAmt GEM amount.
+     */
+    function gemToDai(uint256 gemAmt) external view returns (uint256) {
+        uint256 gemAmt18 = mul(gemAmt, to18ConvertionFactor);
+        uint256 fee = mul(gemAmt18, psm.tin()) / WAD;
+        return sub(gemAmt18, fee);
+    }
+
+    /**
+     * @notice Swaps the specified amount of GEM into DAI through the PSM and push it into the `to` address.
+     * @param amt GEM amount.
+     */
+    function _doPush(uint256 amt) internal {
+        uint256 prevDaiBalance = dai.balanceOf(address(this));
+
+        psm.sellGem(address(this), amt);
+
+        uint256 daiBalance = dai.balanceOf(address(this));
+        uint256 daiPushAmt = sub(daiBalance, prevDaiBalance);
```

Comment:
is this function needed? I don't think it means much as the psm fees are treated as part of the payback, so even if the resulting dai is smaller than the USDC amount the operator shouldn't care.

Reply:
done